src/aegis_ai/osidb_bot/bot.py

```
@@ -451,53 +483,72 @@ async def process_cve(self, cve: CVEID) -> None:
                 read_only=self.read_only,
+                retry_list=self.retry_list,
+                on_failure=self.schedule_retry,
             )
-            self.pending[flaw_updater.state()] = True  # mark as pending
+            if not self.retrying_failed:
+                self.pending[flaw_updater.position()] = True  # mark as pending
             await flaw_updater.do()
 
         except RuntimeError as e:
             # something has failed
             logger.warning(f"{cve}: {str(e)}")
```

Ruff (0.15.20)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)